WordCountAlgo/WordFrequency.quorum

```diff
@@ -44,19 +84,55 @@ class WordFrequency
 
         repeat while wordItr:HasNext()
             text word = wordItr:Next()
-            check
-                integer count = WordCountMap:GetValue(word)
-                count = count + 1
-                WordCountMap:Set(word, count)
-            detect e is UndefinedObjectError
-                WordCountMap:Set(word, 1)
+            // add words that are not stopwords
+            if not StopWords:HasKey(word)
+                check
+                    integer count = WordCountMap:GetValue(word)
+                    count = count + 1
+                    WordCountMap:Set(word, count)
+                detect e is UndefinedObjectError
+                    WordCountMap:Set(word, 1)
+                end
             end
         end
 
         inFile:Close()
-
         return WordCountMap
     end
+    public action CalculateFromArray(Array<text> TextToBeClassified) returns HashTable<text,integer>
+    Iterator<text> OuterItr = TextToBeClassified:GetIterator()
+        repeat while OuterItr:HasNext()
+            text section = OuterItr:Next()
+            Array<text> Words = section:Split(" ")
+            Iterator<text> WordItr = Words:GetIterator()
+                repeat while WordItr:HasNext()
+                    text word = WordItr:Next()
+                    check
+                        integer count = WordCountMap:GetValue(word)
+                        count = count + 1
+                        WordCountMap:Set(word, count)
+                    detect e is UndefinedObjectError
+                        WordCountMap:Set(word, 1)
+                    end                     
+            
+                end
+        end
+        return WordCountMap
+    end
+    // Optional: Function to remove stopwords from WordCountMap
+    public action RemoveStopWords
+        Array<text> keys = WordCountMap:CopyToKeyArray()
+        
+        integer i = 0
+        repeat while i < keys:GetSize()
+            text word = keys:Get(i)
+            if StopWords:HasKey(word)
+                WordCountMap:RemoveKey(word)
+            end
+            i = i + 1
+        end
+    end
+
 
     public action GetTotalWords returns integer
         Iterator<text> wordMapItr = WordCountMap:GetKeyIterator()
```

Comment:
can we add a class variable that tracks how many unique words have been added? Then we can get rid of this function and just add a getter to return it.